### PR TITLE
Contribute initial sample code

### PR DESCRIPTION
**Pull request by julicjung**
*Description of changes:*
- The initial sample code

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .mvn/wrapper/MavenWrapperDownloader.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2007-present the original author or authors.
```
Comment (fbdo) [inline .mvn/wrapper/MavenWrapperDownloader.java]: As this repository includes third party content we need to create a attribution document (usually named THIRD-PARTY-LICENSES, THIRD-PARTY-NOTICES, or just THIRD-PARTY) to comply with the license conditions of that third-party content.

Please check an example here: https://github.com/aws-samples/aws-iot-edukit-examples/blob/main/THIRD-PARTY-LICENSES.txt

For used as part of package manager config files (package.json, requirements.txt, pom.xml etc) it is not necessary to add add them to the attribution file.

File: cdk/bin/app-config-app.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env node
```
Comment (fbdo) [inline cdk/bin/app-config-app.ts:4]: Let's add the shorter license header as in:

```javascript
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: MIT-0
```

File: cdk/lib/app-config-stack.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import * as cdk from '@aws-cdk/core';
```
Comment (fbdo) [inline cdk/lib/app-config-stack.ts:4]: Let's add the shorter license header as in:

```javascript
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: MIT-0
```

File: cdk/package.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "cdk",
+  "version": "0.1.0",
```
Comment (fbdo) [inline cdk/package.json:3]: Please add a license element as explained in https://docs.npmjs.com/cli/v8/configuring-npm/package-json#license, for the MIT-0 license as in:

```json
{
  "license" : "MIT-0"
}
```

File: mvnw [hunk 5]
```diff
@@ -0,0 +1,322 @@
+#!/bin/sh
```
Comment (fbdo) [inline mvnw]: As this repository includes third party content we need to create a attribution document (usually named THIRD-PARTY-LICENSES, THIRD-PARTY-NOTICES, or just THIRD-PARTY) to comply with the license conditions of that third-party content.

Please check an example here: https://github.com/aws-samples/aws-iot-edukit-examples/blob/main/THIRD-PARTY-LICENSES.txt

For used as part of package manager config files (package.json, requirements.txt, pom.xml etc) it is not necessary to add add them to the attribution file.

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,151 @@
+<?xml version="1.0"?>
+<project
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd"
+  xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.amazon</groupId>
+  <artifactId>appconfig-quarkus</artifactId>
```
Comment (fbdo) [inline pom.xml]: Please add a license element as explained here:

https://maven.apache.org/pom.html#Licenses

**Review by fbdo — COMMENTED** [review]
Please check the license headers, make sure they are all MIT-0. We have a file coming from a different project (mvn) with a different license, so we need to create a THIRD-PARTY-LICENSES.txt file.

File: src/main/java/com/amazon/LoanCalculationRequest.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+ * permit persons to whom the Software is furnished to do so.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.amazon;
```
Comment (fbdo) [inline src/main/java/com/amazon/LoanCalculationRequest.java]: Let's move to a more specific package, as in:

```java
package com.amazon.sample.quarkus
```

File: src/main/java/com/amazon/LoanCalculationResponse.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+ * permit persons to whom the Software is furnished to do so.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.amazon;
```
Comment (fbdo) [inline src/main/java/com/amazon/LoanCalculationResponse.java]: Let's move to a more specific package, as in:

```java
package com.amazon.sample.quarkus
```

File: src/main/java/com/amazon/LoanResource.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+ * permit persons to whom the Software is furnished to do so.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.amazon;
```
Comment (fbdo) [inline src/main/java/com/amazon/LoanResource.java]: Let's move to a more specific package, as in:

```java
package com.amazon.sample.quarkus
```

File: src/main/java/com/amazon/config/source/AwsAppConfigSource.java [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+ * permit persons to whom the Software is furnished to do so.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.amazon.config.source;
```
Comment (fbdo) [inline src/main/java/com/amazon/config/source/AwsAppConfigSource.java]: Let's move to a more specific package, as in:

```java
package com.amazon.sample.quarkus.config
```

File: src/test/java/com/amazon/LoanResourceTest.java [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+ * permit persons to whom the Software is furnished to do so.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.amazon;
```
Comment (fbdo) [inline src/test/java/com/amazon/LoanResourceTest.java]: Let's move to a more specific package, as in:

```java
package com.amazon.sample.quarkus
```

File: src/test/java/com/amazon/NativeLoanResourceIT.java [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+ * permit persons to whom the Software is furnished to do so.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.amazon;
```
Comment (fbdo) [inline src/test/java/com/amazon/NativeLoanResourceIT.java]: Let's move to a more specific package, as in:

```java
package com.amazon.sample.quarkus
```

**Review by fbdo — COMMENTED** [review]
I added some comments on to move code to more specific packages

File: src/test/java/com/amazon/NativeLoanResourceIT.java (hunk 12 above)
Comment (julicjung) [reply src/test/java/com/amazon/NativeLoanResourceIT.java]: Changed in e9d4fff

File: src/test/java/com/amazon/LoanResourceTest.java (hunk 11 above)
Comment (julicjung) [reply src/test/java/com/amazon/LoanResourceTest.java]: Changed in e9d4fff

File: src/main/java/com/amazon/config/source/AwsAppConfigSource.java (hunk 10 above)
Comment (julicjung) [reply src/main/java/com/amazon/config/source/AwsAppConfigSource.java]: Changed in e9d4fff

File: src/main/java/com/amazon/LoanResource.java (hunk 9 above)
Comment (julicjung) [reply src/main/java/com/amazon/LoanResource.java]: Changed in e9d4fff

File: src/main/java/com/amazon/LoanCalculationResponse.java (hunk 8 above)
Comment (julicjung) [reply src/main/java/com/amazon/LoanCalculationResponse.java]: Changed in e9d4fff

File: src/main/java/com/amazon/LoanCalculationRequest.java (hunk 7 above)
Comment (julicjung) [reply src/main/java/com/amazon/LoanCalculationRequest.java]: Changed in e9d4fff

File: pom.xml (hunk 6 above)
Comment (julicjung) [reply pom.xml]: Added license MIT-0 in e9d4fff

File: mvnw (hunk 5 above)
Comment (julicjung) [reply mvnw]: Maven wrapper removed in e9d4fff, hence no THIRD PARTY file

File: cdk/package.json (hunk 4 above)
Comment (julicjung) [reply cdk/package.json:3]: Added in e9d4fff

File: cdk/lib/app-config-stack.ts (hunk 3 above)
Comment (julicjung) [reply cdk/lib/app-config-stack.ts:4]: Added in e9d4fff

File: cdk/bin/app-config-app.ts (hunk 2 above)
Comment (julicjung) [reply cdk/bin/app-config-app.ts:4]: Added in e9d4fff

File: .mvn/wrapper/MavenWrapperDownloader.java (hunk 1 above)
Comment (julicjung) [reply .mvn/wrapper/MavenWrapperDownloader.java]: File removed in e9d4fff

**Review by fbdo — APPROVED** [review]
All changes approved, merging changes.

**Pull request merged**